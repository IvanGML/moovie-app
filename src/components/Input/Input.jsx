import React from 'react';
import styles from './Input.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import setNewMoviesList from '../../redux/actions/setNewMoviesList';
import fetchMovies from '../../redux/actions/fetchMovies';
import getSearchValue from '../../redux/actions/getSearchValue';

export class Input extends React.Component {
  constructor(props) {
    super(props);
    this.state = { inputText: '' };
  }

  setValue = (e) => {
    const inputText = e.target.value;
    const result = this.props.movies.data.filter((item) => {
      const currentItem = item.title.toLowerCase();
      return currentItem.includes(inputText.toLowerCase());
    });
    this.setState({ inputText },()=>this.props.setNewMoviesList(result));
    this.props.getSearchValue(inputText);
  }

  render() {
    return (
      <input type="text" 
             value={this.state.inputText} 
             onChange={this.setValue}
             className={styles.inputStyles} />
    );
  }
}

const mapStateToProps = (state) => {
  return {
    movies: state ? state.movies.moviesAll : {},
    searchFlag: state.searchFlag,
  };
};

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setNewMoviesList,
    fetchMovies,
    getSearchValue,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Input);
