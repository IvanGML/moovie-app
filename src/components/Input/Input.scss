$main-color: #f85064;

.inputStyles {
  height: 50px;
  line-height: 50px;
  background-color: rgba($color: #000000, $alpha: .7);
  border: none;
  border-bottom: 2px solid rgba($color: $main-color, $alpha: .7);
  width: 100%;
  color: #fafafa;
  padding: 0px 25px;
  box-sizing: border-box;
}
