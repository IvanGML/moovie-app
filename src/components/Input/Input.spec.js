import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from './Input';


test('< Input />', () => {
  const component = renderer.create(
    <Input />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
