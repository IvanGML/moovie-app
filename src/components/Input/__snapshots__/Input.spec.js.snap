// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Input /> 1`] = `
<input
  className="inputStyles"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
